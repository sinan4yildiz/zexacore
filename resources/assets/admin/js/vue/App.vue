<template>
  <div class="min-h-screen">

    <!-- If connection is gone -->
    <Offline></Offline>

    <!-- Header -->
    <Header @handleNavigation="navigationStatus = $event"></Header>

    <!-- Wrapper -->
    <div class="grid grid-cols-16">

      <!-- Sidebar -->
      <Sidebar class="col-span-full lg:col-span-4 xl:col-span-3 -mt-2 lg:mt-0 px-4 2xl:pl-10 lg:pr-10 2xl:pr-16 lg:block" v-bind:class="{'hidden': !navigationStatus}"></Sidebar>

      <!-- Right hand side -->
      <div class="col-span-full lg:col-span-12 xl:col-span-13 2xl:col-span-12 px-4 lg:pl-0 2xl:px-0">

        <!-- Page content -->
        <main>
          <RouterView></RouterView>
        </main>

        <!-- Footer -->
        <Footer></Footer>
      </div>
    </div>

    <!-- Snackbar -->
    <Snackbar></Snackbar>

    <!-- SVG library -->
    <SVGLibrary></SVGLibrary>
  </div>
</template>

<script>
import {mapActions, mapGetters} from 'vuex'
import Header from "./components/Header";
import Sidebar from "./components/Sidebar";
import Footer from "./components/Footer";
import Dropdown from "./components/elements/Dropdown";
import SVGLibrary from "./components/SVGLibrary";
import Offline from "./components/elements/Offline";
import Snackbar from "./components/elements/Snackbar";

export default {
  name: "App",

  data() {
    return {
      navigationStatus: false,
      metaTitle: this.$t('common.brand'),
    }
  },

  computed: {
    ...mapGetters('Languages', ['languages']),
  },

  methods: {
    ...mapActions('Languages', ['fetchLanguages']),

    setMetaTitle: function (to) {
      if(this.$route.matched.length)
        this.metaTitle = to.meta.title || this.$route.matched[0].meta.title
    },

    // Prefetch the languages for translatable modules
    prefetchLanguages: function () {
      if(this.$route.meta.translatable && !this.languages.data) {
        this.fetchLanguages()
      }
    }
  },

  watch: {
    $route(to) {
      this.setMetaTitle(to)
      this.prefetchLanguages()
    },

    metaTitle() {
      document.title = this.metaTitle
    }
  },

  created() {
    this.setMetaTitle(this.$route)
    this.prefetchLanguages()
  },

  components: {
    Header,
    Sidebar,
    Footer,
    Dropdown,
    SVGLibrary,
    Offline,
    Snackbar,
  }
}
</script>