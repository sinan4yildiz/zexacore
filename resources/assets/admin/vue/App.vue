<template>
  <div class="min-h-screen">

    <!-- If connection is gone -->
    <Offline></Offline>

    <!-- Checking for authentication -->
    <div v-if="verifying" class="flex flex-col justify-center items-center w-full min-h-screen">
      <svg class="w-5 sm:w-6 h-5 sm:h-6 text-blue-600 animate-spin-fast">
        <use xlink:href="#icon-loading"></use>
      </svg>
      <div class="mt-4 text-sm font-light text-gray-600">{{ $t('common.please_wait') }}</div>
    </div>

    <!-- Authenticated layout -->
    <Authenticated v-else-if="isAuthenticated"></Authenticated>

    <!-- Unauthenticated layout -->
    <Unauthenticated v-else></Unauthenticated>

    <!-- Snackbar -->
    <Snackbar></Snackbar>

    <!-- SVG library -->
    <SVGLib></SVGLib>
  </div>
</template>

<script>
import { mapActions, mapGetters } from 'vuex';
import Unauthenticated from '@/layouts/Unauthenticated';
import Snackbar from '@/components/Elements/Snackbar';
import Authenticated from '@/layouts/Authenticated';
import Offline from '@/components/Elements/Offline';
import SVGLib from '@/components/SVGLib';

export default {
  name: 'App',

  data() {
    return {
      verifying: true,
    };
  },

  computed: {
    ...mapGetters('Auth', ['isAuthenticated']),
  },

  methods: {
    ...mapActions('App', ['initApp']),
    ...mapActions('Auth', ['verifyAuth']),

    setMetaTitle(route) {
      if (this.$route.matched.length) {
        this.metaTitle = route.meta.title || this.$route.matched[0].meta.title;

        document.title = this.metaTitle;
      }
    },
  },

  watch: {
    $route(to) {
      this.setMetaTitle(to);
    },
  },

  beforeMount() {
    /*
     * Init app
     *
     * */
    this.initApp();

    /*
     * Verify authentication
     *
     * */
    this.verifyAuth()
      .finally(() => {
        this.verifying = false;
      });
  },

  mounted() {
    this.setMetaTitle(this.$route);
  },

  components: {
    Offline,
    Snackbar,
    Authenticated,
    Unauthenticated,
    SVGLib,
  },
};
</script>
