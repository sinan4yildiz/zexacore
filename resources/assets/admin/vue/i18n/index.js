import Vue from 'vue';
import VueI18n from 'vue-i18n';

import en from './lang/en.json';
/* import tr from './lang/tr.json'; */

Vue.use(VueI18n);

const i18n = new VueI18n({
  locale: 'en',
  messages: { en },
});

export default i18n;
