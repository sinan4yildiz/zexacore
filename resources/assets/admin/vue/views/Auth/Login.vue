<template>
  <section class="flex justify-center items-center px-8 pt-6 pb-24 min-h-screen">
    <div class="w-full max-w-md">

      <!-- Logo -->
      <svg class="block mx-auto mb-8 w-28 sm:w-34 h-24 sm:h-28">
        <use xlink:href="#logo-vertical"></use>
      </svg>

      <form @submit.prevent="login" class="py-10 sm:py-16 px-6 sm:px-12 bg-white rounded-lg shadow-sm" method="post">

        <!-- Email -->
        <div class="mb-4">
          <Input name="email" type="email" :label="$t('label.email')" :placeholder="$t('placeholder.email')" @input="form.email = $event" :errors="errors"/>
        </div>

        <!-- Password -->
        <div class="mb-4">
          <Input name="password" type="password" :label="$t('label.password')" placeholder="●●●●●" @input="form.password = $event" :errors="errors"/>
        </div>

        <!-- Remember and forgotten password -->
        <div class="flex justify-between items-center mb-4">
          <div class="flex items-center">
            <input type="checkbox" name="remember" id="remember" class="w-4 h-4 text-blue-600 border-gray-400 focus:border-blue-600 transition duration-150 ease-in-out form-checkbox focus:shadow-outline-blue">
            <label for="remember" class="block ml-2 text-sm leading-5 text-gray-600 hover:text-gray-700 select-none">
              {{ $t('auth.login.remember_me') }}
            </label>
          </div>
          <div class="text-sm leading-5">
            <RouterLink :to="{hash: '#'}" class="text-sm font-medium text-blue-600 hover:text-blue-500 focus:underline transition duration-150 ease-in-out focus:outline-none">
              {{ $t('auth.login.forgotten_password') }}
            </RouterLink>
          </div>
        </div>

        <!-- Submit button -->
        <Button type="submit" theme="blue" size="large-full" :label="$t('auth.login.button')" ref="loginButton"/>
      </form>

      <!-- View website -->
      <div class="mt-5 ml-2">
        <a :href="config.url.base" class="inline-flex items-center text-sm text-gray-500 hover:text-blue-600 transition duration-150 ease-in-out">
          <svg class="mr-2 w-6 h-4 text-gray-500">
            <use xlink:href="#icon-arrow-left"></use>
          </svg>
          {{ $t('common.view_website') }}
        </a>
      </div>
    </div>
  </section>
</template>

<script>
import { mapActions, mapGetters } from 'vuex';
import Button from '@/components/Form/Button';
import Input from '@/components/Form/Input';

export default {
  name: 'Login',

  data() {
    return {
      form: {},
      errors: {},
    };
  },

  computed: {
    ...mapGetters('App', ['config']),
  },

  methods: {
    ...mapActions('Auth', ['userLogin']),

    login() {
      this.$refs.loginButton.loading = true;

      this.userLogin(this.form)
        .catch((error) => {
          if (error.response.status === 401) {
            this.$snackbar(error.response.data.error, 'error');
          } else {
            this.errors = error.response.data.errors;
          }

          _.delay(() => {
            this.$refs.loginButton.loading = false;
          }, 500);
        });
    },
  },

  components: {
    Button,
    Input,
  },
};
</script>
