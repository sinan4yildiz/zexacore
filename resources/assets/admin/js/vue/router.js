import Vue from 'vue'
import VueRouter from 'vue-router'
import i18n from "./i18n";

import DashboardIndex from './views/Dashboard/Index'

import ArticlesIndex from './views/Content/Articles/Index.vue'
import ArticlesCreate from './views/Content/Articles/Create.vue'
import ArticlesEdit from './views/Content/Articles/Edit.vue'

import PagesIndex from './views/Content/Pages/Index.vue'
import PagesCreate from './views/Content/Pages/Create.vue'
import PagesEdit from './views/Content/Pages/Edit.vue'

import UploadsIndex from './views/Content/Uploads/Index.vue'
import MessagesIndex from './views/Messages/Index.vue'

import CategoriesIndex from './views/Structure/Categories/Index.vue'
import CategoriesCreate from './views/Structure/Categories/Create.vue'
import CategoriesEdit from './views/Structure/Categories/Edit.vue'

import ContentTypesIndex from './views/Structure/ContentTypes/Index.vue'
import ContentTypesCreate from './views/Structure/ContentTypes/Create.vue'
import ContentTypesEdit from './views/Structure/ContentTypes/Edit.vue'

import UsersIndex from './views/People/Users/Index.vue'
import ActivitiesIndex from './views/People/Activities/Index.vue'
import SlugsIndex from './views/Tools/Slugs/Index'
import RedirectionsIndex from './views/Tools/Redirections/Index'
import NotFoundsIndex from "./views/Tools/NotFounds/Index"
import SettingsIndex from './views/System/Settings/Index'
import LanguagesIndex from './views/System/Languages/Index'
import MaintenanceIndex from './views/System/Maintenance/Index'
import NotFound from './views/Errors/NotFound'

Vue.use(VueRouter)

export default new VueRouter({
    base: window.data.url.admin.replace(window.location.origin, ''),
    routes: [

        /*
        * Dashboard
        *
        * */
        {
            path: '/',
            name: 'dashboard',
            component: DashboardIndex,
            meta: {
                title: i18n.t('dashboard.heading')
            }
        },


        /*
        * Content
        *
        * */
        {
            path: '/articles',
            name: 'articles',
            component: ArticlesIndex,
            meta: {
                title: i18n.t('articles.heading.index'),
                translatable: true
            },
        },
        {
            path: '/articles/create/:language?',
            name: 'articles.create',
            component: ArticlesCreate,
            meta: {
                title: i18n.t('articles.heading.create'),
                breadcrumb: i18n.t('common.create'),
                translatable: true
            },
        },
        {
            path: '/articles/edit/:id',
            name: 'articles.edit',
            component: ArticlesEdit,
            meta: {
                title: i18n.t('articles.heading.edit'),
                breadcrumb: i18n.t('common.edit'),
                translatable: true
            },
        },
        {
            path: '/pages',
            name: 'pages',
            component: PagesIndex,
            meta: {
                title: i18n.t('pages.heading.index'),
                translatable: true
            },
        },
        {
            path: '/pages/create/:language?',
            name: 'pages.create',
            component: PagesCreate,
            meta: {
                title: i18n.t('pages.heading.create'),
                breadcrumb: i18n.t('common.create'),
                translatable: true
            },
        },
        {
            path: '/pages/edit/:id',
            name: 'pages.edit',
            component: PagesEdit,
            meta: {
                title: i18n.t('pages.heading.edit'),
                breadcrumb: i18n.t('common.edit'),
                translatable: true
            },
        },

        // Uploads
        {
            path: '/uploads',
            name: 'uploads',
            component: UploadsIndex,
            meta: {
                title: i18n.t('uploads.heading.index'),
            },
        },


        /*
        * Messages
        *
        * */
        {
            path: '/messages',
            name: 'messages',
            component: MessagesIndex,
            meta: {
                title: i18n.t('messages.heading.index'),
            },
        },


        /*
        * Structure
        *
        * */

        // Categories
        {
            path: '/categories',
            name: 'categories',
            component: CategoriesIndex,
            meta: {
                title: i18n.t('categories.heading.index'),
                translatable: true
            },
        },
        {
            path: '/categories/create',
            name: 'categories.create',
            component: CategoriesCreate,
            meta: {
                title: i18n.t('categories.heading.create'),
                breadcrumb: i18n.t('common.create'),
                translatable: true
            },
        },
        {
            path: '/categories/edit/:id/:language?',
            name: 'categories.edit',
            component: CategoriesEdit,
            meta: {
                title: i18n.t('categories.heading.edit'),
                breadcrumb: i18n.t('common.edit'),
                translatable: true
            },
        },

        // Content types
        {
            path: '/content-types',
            name: 'content_types',
            component: ContentTypesIndex,
            meta: {
                title: i18n.t('content_types.heading.index'),
                translatable: true
            },
        },
        {
            path: '/content-types/create',
            name: 'content_types.create',
            component: ContentTypesCreate,
            meta: {
                title: i18n.t('content_types.heading.create'),
                breadcrumb: i18n.t('common.create'),
                translatable: true
            },
        },
        {
            path: '/content-types/edit/:id/:language?',
            name: 'content_types.edit',
            component: ContentTypesEdit,
            meta: {
                title: i18n.t('content_types.heading.edit'),
                breadcrumb: i18n.t('common.edit'),
                translatable: true
            },
        },


        /*
        * People
        *
        * */
        {
            path: '/users',
            name: 'users',
            component: UsersIndex,
            props: true,
            meta: {
                title: i18n.t('users.heading.index')
            }
        },
        {
            path: '/activities',
            name: 'activities',
            component: ActivitiesIndex,
            meta: {
                title: i18n.t('activities.heading.index')
            },
        },


        /*
        * Tools
        *
        * */
        {
            path: '/url-aliases',
            name: 'slugs',
            component: SlugsIndex,
            meta: {
                title: i18n.t('slugs.heading.index')
            }
        },
        {
            path: '/redirections',
            name: 'redirections',
            component: RedirectionsIndex,
            meta: {
                title: i18n.t('redirections.heading.index')
            },
        },
        {
            path: '/not-founds',
            name: 'not_founds',
            component: NotFoundsIndex,
            meta: {
                title: i18n.t('not_founds.heading.index')
            },
        },


        /*
        * System
        *
        * */
        {
            path: '/settings',
            name: 'settings',
            component: SettingsIndex,
            meta: {
                title: i18n.t('settings.heading.index')
            },
        },
        {
            path: '/languages',
            name: 'languages',
            component: LanguagesIndex,
            meta: {
                title: i18n.t('settings.heading.index')
            },
        },
        {
            path: '/maintenance',
            name: 'maintenance',
            component: MaintenanceIndex,
            meta: {
                title: i18n.t('maintenance.heading.index')
            },
        },

        {
            path: '*',
            name: 'not_found',
            component: NotFound,
            meta: {
                title: i18n.t('not_found.heading')
            },
        }
    ],
    mode: 'history'
});