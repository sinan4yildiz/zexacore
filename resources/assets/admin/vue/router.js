/*
 * Libraries
 *
 * */
import Vue from 'vue';
import VueRouter from 'vue-router';
import i18n from '@/i18n';

/*
 * Views
 *
 * */
import LoginIndex from '@/views/Auth/Login';
import DashboardIndex from '@/views/Dashboard/Index';

import ArticlesIndex from '@/views/Content/Articles/Index';
import ArticlesCreate from '@/views/Content/Articles/Create';
import ArticlesEdit from '@/views/Content/Articles/Edit';

import PagesIndex from '@/views/Content/Pages/Index';
import PagesCreate from '@/views/Content/Pages/Create';
import PagesEdit from '@/views/Content/Pages/Edit';

import UploadsIndex from '@/views/Content/Uploads/Index';
import MessagesIndex from '@/views/Messages/Index';

import CategoriesIndex from '@/views/Structure/Categories/Index';
import CategoriesCreate from '@/views/Structure/Categories/Create';
import CategoriesEdit from '@/views/Structure/Categories/Edit';

import ContentTypesIndex from '@/views/Structure/ContentTypes/Index';
import ContentTypesCreate from '@/views/Structure/ContentTypes/Create';
import ContentTypesEdit from '@/views/Structure/ContentTypes/Edit';

import UsersIndex from '@/views/People/Users/Index';
import ActivitiesIndex from '@/views/People/Activities/Index';
import SlugsIndex from '@/views/Tools/Slugs/Index';
import RedirectionsIndex from '@/views/Tools/Redirections/Index';
import NotFoundsIndex from '@/views/Tools/NotFounds/Index';
import SettingsIndex from '@/views/System/Settings/Index';
import LanguagesIndex from '@/views/System/Languages/Index';
import MaintenanceIndex from '@/views/System/Maintenance/Index';
import NotFound from '@/views/Errors/NotFound';

/*
 * Routes
 *
 * */
const routes = [

  /*
   * Auth
   *
   * */
  {
    path: '/login',
    name: 'login',
    component: LoginIndex,
    meta: {
      title: i18n.t('auth.login.heading'),
      requiresAuth: false,
    },
  },

  /*
   * Dashboard
   *
   * */
  {
    path: '/',
    name: 'dashboard',
    component: DashboardIndex,
    meta: {
      title: i18n.t('dashboard.heading'),
      requiresAuth: true,
    },
  },

  /*
   * Content
   *
   * */
  {
    path: '/articles',
    name: 'articles',
    component: ArticlesIndex,
    meta: {
      title: i18n.t('articles.heading.index'),
      translatable: true,
      requiresAuth: true,
    },
  },
  {
    path: '/articles/create/:language?',
    name: 'articles.create',
    component: ArticlesCreate,
    meta: {
      title: i18n.t('articles.heading.create'),
      breadcrumb: i18n.t('common.create'),
      translatable: true,
      requiresAuth: true,
    },
  },
  {
    path: '/articles/edit/:id',
    name: 'articles.edit',
    component: ArticlesEdit,
    meta: {
      title: i18n.t('articles.heading.edit'),
      breadcrumb: i18n.t('common.edit'),
      translatable: true,
      requiresAuth: true,
    },
  },
  {
    path: '/pages',
    name: 'pages',
    component: PagesIndex,
    meta: {
      title: i18n.t('pages.heading.index'),
      translatable: true,
      requiresAuth: true,
    },
  },
  {
    path: '/pages/create/:language?',
    name: 'pages.create',
    component: PagesCreate,
    meta: {
      title: i18n.t('pages.heading.create'),
      breadcrumb: i18n.t('common.create'),
      translatable: true,
      requiresAuth: true,
    },
  },
  {
    path: '/pages/edit/:id',
    name: 'pages.edit',
    component: PagesEdit,
    meta: {
      title: i18n.t('pages.heading.edit'),
      breadcrumb: i18n.t('common.edit'),
      translatable: true,
      requiresAuth: true,
    },
  },

  // Uploads
  {
    path: '/uploads',
    name: 'uploads',
    component: UploadsIndex,
    meta: {
      title: i18n.t('uploads.heading.index'),
      requiresAuth: true,
    },
  },

  /*
   * Messages
   *
   * */
  {
    path: '/messages',
    name: 'messages',
    component: MessagesIndex,
    meta: {
      title: i18n.t('messages.heading.index'),
      requiresAuth: true,
    },
  },

  /*
   * Structure
   *
   * */

  // Categories
  {
    path: '/categories',
    name: 'categories',
    component: CategoriesIndex,
    meta: {
      title: i18n.t('categories.heading.index'),
      translatable: true,
      requiresAuth: true,
    },
  },
  {
    path: '/categories/create',
    name: 'categories.create',
    component: CategoriesCreate,
    meta: {
      title: i18n.t('categories.heading.create'),
      breadcrumb: i18n.t('common.create'),
      translatable: true,
      requiresAuth: true,
    },
  },
  {
    path: '/categories/edit/:id/:language?',
    name: 'categories.edit',
    component: CategoriesEdit,
    meta: {
      title: i18n.t('categories.heading.edit'),
      breadcrumb: i18n.t('common.edit'),
      translatable: true,
      requiresAuth: true,
    },
  },

  // Content types
  {
    path: '/content-types',
    name: 'content_types',
    component: ContentTypesIndex,
    meta: {
      title: i18n.t('content_types.heading.index'),
      translatable: true,
      requiresAuth: true,
    },
  },
  {
    path: '/content-types/create',
    name: 'content_types.create',
    component: ContentTypesCreate,
    meta: {
      title: i18n.t('content_types.heading.create'),
      breadcrumb: i18n.t('common.create'),
      translatable: true,
      requiresAuth: true,
    },
  },
  {
    path: '/content-types/edit/:id/:language?',
    name: 'content_types.edit',
    component: ContentTypesEdit,
    meta: {
      title: i18n.t('content_types.heading.edit'),
      breadcrumb: i18n.t('common.edit'),
      translatable: true,
      requiresAuth: true,
    },
  },

  /*
   * People
   *
   * */
  {
    path: '/users',
    name: 'users',
    component: UsersIndex,
    props: true,
    meta: {
      title: i18n.t('users.heading.index'),
      requiresAuth: true,
    },
  },
  {
    path: '/activities',
    name: 'activities',
    component: ActivitiesIndex,
    meta: {
      title: i18n.t('activities.heading.index'),
      requiresAuth: true,
    },
  },

  /*
   * Tools
   *
   * */
  {
    path: '/url-aliases',
    name: 'slugs',
    component: SlugsIndex,
    meta: {
      title: i18n.t('slugs.heading.index'),
      requiresAuth: true,
    },
  },
  {
    path: '/redirections',
    name: 'redirections',
    component: RedirectionsIndex,
    meta: {
      title: i18n.t('redirections.heading.index'),
      requiresAuth: true,
    },
  },
  {
    path: '/not-founds',
    name: 'not_founds',
    component: NotFoundsIndex,
    meta: {
      title: i18n.t('not_founds.heading.index'),
      requiresAuth: true,
    },
  },

  /*
   * System
   *
   * */
  {
    path: '/settings',
    name: 'settings',
    component: SettingsIndex,
    meta: {
      title: i18n.t('settings.heading.index'),
      requiresAuth: true,
    },
  },
  {
    path: '/languages',
    name: 'languages',
    component: LanguagesIndex,
    meta: {
      title: i18n.t('settings.heading.index'),
      requiresAuth: true,
    },
  },
  {
    path: '/maintenance',
    name: 'maintenance',
    component: MaintenanceIndex,
    meta: {
      title: i18n.t('maintenance.heading.index'),
      requiresAuth: true,
    },
  },

  /*
   * Not found
   *
   * */
  {
    path: '*',
    name: 'not_found',
    component: NotFound,
    meta: {
      title: i18n.t('not_found.heading'),
    },
  },
];

/*
 * Build router
 *
 * */
Vue.use(VueRouter);

const router = new VueRouter({
  base: appConfig.url.admin.replace(window.location.origin, ''),
  routes,
  mode: 'history',
  scrollBehavior(to, from, savedPosition) {
    if (savedPosition) {
      return savedPosition;
    }
    return { x: 0, y: 0 };
  },
});

export default router;
