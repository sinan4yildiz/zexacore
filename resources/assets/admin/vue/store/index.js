import Vue from 'vue';
import Vuex from 'vuex';

/*
* Modules
*
* */
import App from './modules/app';
import Utils from './modules/utils';
import Auth from './modules/auth';
import Dashboard from './modules/dashboard';
import Articles from './modules/Content/articles';
import Pages from './modules/Content/pages';
import Uploads from './modules/Content/uploads';
import Categories from './modules/Structure/categories';
import ContentTypes from './modules/Structure/contentTypes';
import Messages from './modules/messages';
import Users from './modules/People/users';
import Activities from './modules/People/activities';
import Redirections from './modules/Tools/redirections';
import NotFounds from './modules/Tools/notFounds';
import Slugs from './modules/Tools/slugs';
import Languages from './modules/System/languages';
import Settings from './modules/System/settings';
import Maintenance from './modules/System/maintenance';
import Search from './modules/search';

/*
* Binding
*
* */
Vue.use(Vuex);

/*
* Export modules
*
* */
export default new Vuex.Store({
  modules: {
    App,
    Utils,
    Auth,
    Dashboard,
    Articles,
    Pages,
    Uploads,
    Categories,
    ContentTypes,
    Messages,
    Users,
    Activities,
    Redirections,
    NotFounds,
    Slugs,
    Languages,
    Settings,
    Maintenance,
    Search,
  },
  strict: process.env.NODE_ENV !== 'production',
});
