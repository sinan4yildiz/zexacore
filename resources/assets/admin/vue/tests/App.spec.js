import { createLocalVue, mount } from '@vue/test-utils';
import i18n from '@/i18n';
import router from '@/router';
import store from '@/store/index';

/*
* Components
*
* */
import App from '@/App';
import Login from '@/views/Auth/Login';

let wrapper;
const localVue = createLocalVue();

beforeAll(() => {
  wrapper = mount(App, {
    localVue,
    router,
    store,
    i18n,
    mocks: {},
    stubs: {

      // No need this
      SVGLib: true,
    },
  });
});

afterAll(() => {
  wrapper.destroy();
});

describe('App', () => {
  /*
  * Check if the application is initialized without any crash
  *
  * */
  test('The app container should be rendered correctly', async () => {
    // Skip the verifying step
    await wrapper.setData({ verifying: false });

    expect(wrapper).toMatchSnapshot();
  });

  /*
  * Check if the unauthenticated layout rendered
  *
  * */
  test('The authentication component should be rendered correctly', () => {
    expect(wrapper.findAllComponents(Login)).toHaveLength(1);
  });
});
