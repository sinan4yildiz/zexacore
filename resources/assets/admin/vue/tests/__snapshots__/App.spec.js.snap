// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App The app container should be rendered correctly 1`] = `
<div class="min-h-screen">
  <!---->
  <transition-stub enterclass="opacity-0 -translate-y-4" entertoclass="opacity-100 translate-y-0" enteractiveclass="transition ease-out duration-500 transform">
    <section class="flex justify-center items-center px-4 sm:px-6 lg:px-8 pt-6 pb-24 min-h-screen" style="">
      <div class="w-full max-w-md"><svg class="block mx-auto mb-8 w-28 sm:w-34 h-24 sm:h-28">
          <use xlink:href="#logo-vertical"></use>
        </svg>
        <form method="post" class="py-10 sm:py-16 px-6 sm:px-12 bg-white rounded-lg shadow-sm">
          <div class="mb-4">
            <div class="w-full"><label for="email" class="block mb-1 ml-1 text-sm font-medium leading-5 text-gray-700 select-none">Email address</label>
              <div class="relative"><input name="email" id="email" placeholder="user@email.com" type="email" class="block py-3 px-4 w-full text-sm rounded-md border border-gray-400 focus:border-blue-400 shadow-sm transition duration-150 ease-in-out form-input focus:shadow-outline-blue">
                <!---->
              </div>
              <!---->
            </div>
          </div>
          <div class="mb-4">
            <div class="w-full"><label for="password" class="block mb-1 ml-1 text-sm font-medium leading-5 text-gray-700 select-none">Password</label>
              <div class="relative"><input name="password" id="password" placeholder="●●●●●" type="password" class="block py-3 px-4 w-full text-sm rounded-md border border-gray-400 focus:border-blue-400 shadow-sm transition duration-150 ease-in-out form-input focus:shadow-outline-blue"> <button type="button" class="absolute top-0 right-0 bottom-0 py-3 px-4 leading-3 text-gray-600 hover:text-gray-900 focus:outline-none"><svg class="w-5 h-5 transition duration-300 ease-out pointer-events-none">
                    <use xlink:href="#icon-eye"></use>
                  </svg></button></div>
              <!---->
            </div>
          </div>
          <div class="flex justify-between items-center mb-4">
            <div class="flex items-center"><input type="checkbox" name="remember" id="remember" class="w-4 h-4 text-blue-600 border-gray-400 focus:border-blue-600 transition duration-150 ease-in-out form-checkbox focus:shadow-outline-blue"> <label for="remember" class="block ml-2 text-sm leading-5 text-gray-600 hover:text-gray-700 select-none">
                Remember me
              </label></div>
            <div class="text-sm leading-5"><a href="/zexacore/public/admin/login#" class="text-sm font-medium text-blue-600 hover:text-blue-500 focus:underline transition duration-150 ease-in-out focus:outline-none">
                Forgotten password
              </a></div>
          </div> <button type="submit" class="flex align-items-center text-sm font-medium rounded-md whitespace-no-wrap focus:outline-none transition duration-150 ease-in-out select-none py-3 w-full leading-5 justify-center text-white bg-blue-600 border border-blue-600 hover:bg-blue-500 hover:border-blue-500 focus:border-blue-500 focus:shadow-outline-blue shadow-md">
            <!----> Sign in to your account
          </button>
        </form>
        <div class="mt-5 ml-2"><a href="http://localhost/zexacore/public/" class="inline-flex items-center text-sm text-gray-500 hover:text-blue-600 transition duration-150 ease-in-out"><svg class="mr-2 w-6 h-4 text-gray-500">
              <use xlink:href="#icon-arrow-left"></use>
            </svg>
            View website
          </a></div>
      </div>
    </section>
  </transition-stub>
  <transition-stub enterclass="opacity-0 translate-y-2" leaveclass="opacity-100" entertoclass="opacity-100 translate-y-0" leavetoclass="opacity-0 -translate-y-2" enteractiveclass="transition ease-out duration-150" leaveactiveclass="transition ease-in duration-150">
    <!---->
  </transition-stub>
  <svglib-stub></svglib-stub>
</div>
`;
